docs/GETTINGSTARTED.md

```diff
@@ -0,0 +1,182 @@
+
+`Note:` If deploying an ingress controller using [Helm on GCP](https://github.com/GoogleCloudPlatform/community/blob/master/tutorials/nginx-ingress-gke/index.md#deploy-nginx-ingress-controller-with-rbac-enabled), it will likely be necessary for the `--set controller.publishService.enabled=true` flag to be set, so that the created ingress uses the ingress controller's IP address/hostname. The ingress IP address should match the ingress controller's, as this is the IP address that Yggdrasil will use to generate config for envoy.
+
+Assuming we have a simple HTTP web service called 'hello-world', we can apply the following 'hello-world' ingress resource:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: hello-world
+  namespace: default
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    yggdrasil.uswitch.com/healthcheck-path: /healthz
+    yggdrasil.uswitch.com/timeout: 30s
+spec:
+  rules:
+  - host: example.com
+    http:
+      paths:
+      - backend:
+          serviceName: hello-world
+          servicePort: 80
+```
+
+Once the resource has been created, we should see the ingress controller's IP address or hostname when fetching the ingress:
+
+```console
+$ kubectl get ingress hello-world
+NAME          HOSTS              ADDRESS        PORTS   AGE
+hello-world   example.com        192.168.0.10   80      1h
+```
+
+Double check that this matches the ingress controller's external address:
+
+```console
+$ kubectl get svc nginx-ingress-controller
+NAME                       TYPE           CLUSTER-IP   EXTERNAL-IP    PORT(S)                      AGE
+nginx-ingress-controller   LoadBalancer   10.10.10.10  192.168.0.10   80:30757/TCP,443:31061/TCP   1h
+```
+
+We can verify that the ingress is working correctly by cURLing the ingress controller's IP address:
+
+```console
+$ curl -H Host:example.com http://192.168.0.10
+Hello world!
+```
+
+## Configure Yggdrasil
+With our ingress working correctly, we can now setup Yggdrasil. Pull the latest development docker image with the following command:
+
+```console
+$ docker pull quay.io/uswitch/yggdrasil:devel
+```
+
+Next, we will setup a config file for Yggdrasil so we can retrieve ingress details from our Kubernetes cluster's API. Consider the following Yggdrasil config:
+
+```yaml
+{
+  "nodeName": "envoy-node",
+  "ingressClasses": ["nginx"],
+  "clusters": [
+    {
+      "token": "kubeApiToken",
+      "apiServer": "https://kube.api.server:<port>",
+      "ca": "ca.crt"
+    }
+  ]
+}
+```
+
+Where:
+* `nodeName` is the name we will give our envoy node(s)
+* `ingressClasses` is a list of the ingress classes that Yggdrasil will look for
+* `clusters` is a list of Kubernetes clusters, where:
+  * `token` is the Kube token of a service account that is able to get ingress resources
+  * `apiServer` is the address of the Kube api server
+  * `ca` is the Kube API CA certificate
+
```

Comment:
I think it would be useful to included a service account and the related RBAC for this to work